### DISCUSSION TITLE
Edit options for block component directly inside the editor without opening a popup

### DISCUSSION THREAD

**Discussion by zanhk** [Thinkmill/keystatic#998] [Ideas]
I was thinking that for the block component as there is no need for child components anyway would be more straighforward to ble able to edit the content directly without having to open a popup, so from this

![image](https://github.com/Thinkmill/keystatic/assets/17980844/5f71f6c6-e808-4fd5-91f8-7668ae17f64f)

to this

![image](https://github.com/Thinkmill/keystatic/assets/17980844/fa34911e-b616-48a3-adcd-1dc049e48337)

What do you think?

**Comment by simonswiss** [comment]
That's indeed an interesting discussion point.

I know @JedWatson was hoping that more folks using those blocks would expose the need for different editing patterns — and this one maybe be one of those.

I agree that a config to switch between editing in modal or "inline" would be cool, as long as the implementation doesn't complicate things. 

Hiding the fields in a modal sorta forces you to provide a basic `ContentView` to make this information visible in the content field — so there's definitely value in being able to decide to see the fields inline.

**Comment by Matthijz98** [comment]
It feels the most out of place when you have a array of tags and you need to open a modal to edit or add a new tag.

**Comment by zanhk** [comment]
The case is this one

![image](https://github.com/Thinkmill/keystatic/assets/17980844/b382f43d-7744-4133-aeeb-44d0e7117279)

The customer is confused and doesn't know where to change the text he want to change, if he see insted the component "inline" he would quickly identify the text that he want to change.
